### PR TITLE
Fixed registering runtime Type

### PR DESCRIPTION
**Pull request by Zimnx**

### CODE REVIEW

File: extension/golang/environment.go [hunk 1]
```diff
@@ -48,17 +49,16 @@ type Environment struct {
 
 	// golang extension environment
 	extEnvironment goext.Environment
-	resourceTypes  map[string]reflect.Type
+	runtimeType    reflect.Type
```
Comment (borys-cl) [inline extension/golang/environment.go]: @Zimnx  why did you remove the map? Original idea was to be able to register multiple types. Won't we need that?

Comment (Zimnx) [inline extension/golang/environment.go]: You're right. Yesterday I thought that there is 1-1 mapping of schema to env. That's obviously wrong. There is still bug there. During type registration, type name is used as key in map. Later when concrete type is needed to instantiate resource, schema.Titile is used.

**Comment by Zimnx** [comment]
@borys-cl please take a look at this
